src/awkward/_v2/contents/content.py

```diff
@@ -214,10 +214,24 @@ def __getitem__(self, where):
                 return self.__getitem__((where,))
 
             elif isinstance(where, tuple):
+
                 if len(where) == 0:
                     return self
+
+                prepare_tuple = []
+                for i in range(len(where)):
+                    x = self._prepare_tuple_item(where[i])
+                    if isinstance(x, list) and len(x) == 0:
+                        pass
```

Comment:
Since `x` is the output of `_prepare_tuple_item`, `isinstance(x, list)` means that `x` is a list of strings, slicing fields. Skipping it would be equivalent to taking all of the fields (leaving the RecordArray as-is), but `x == []` would be equivalent to taking none of the fields (replacing the RecordArray with a RecordArray of the same `length` but no fields). Thus, this item should not be skipped; if there's an error in `_getitem_fields` that makes it behave badly, that error should be fixed in `_getitem_fields`. Which test instance motivated this?

Reply:
test_0008_v2-slices-and-getitem.py:74
test_0021_v2-emptyarray.py::test_getitem:39